functions/deploy/main.py

```
@@ -26,8 +26,14 @@ class Door43Deployer(object):
     def __init__(self, cdn_bucket, door43_bucket):
         self.cdn_bucket = cdn_bucket
         self.door43_bucket = door43_bucket
-        self.cdn_handler = S3Handler(cdn_bucket)
-        self.door43_handler = S3Handler(door43_bucket)
+
+        self.cdn_handler = S3Handler(cdn_bucket) if cdn_bucket else None
+        self.door43_handler = S3Handler(door43_bucket) if door43_bucket else None
```

What is the purpose of this change? The handlers are required, so this seems to just ignore an error.

---

@phillip-hopper - as written the method was not very easy to unit test.  I was trying to make minimal changes.  I was trying to work around exceptions being thrown in init method.  I will look for another approach.

---

@PhotoNomad0 We started passing the S3Handler class to the functions so it was easy to mock during testing. See https://github.com/unfoldingWord-dev/d43-catalog/blob/develop/tests/signing/test_signing.py.

---

@phillip-hopper - thanks, will take a look as I have time.